lib/src/main/java/com/diffplug/spotless/JavaVersion.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless;
+
+/**
+ * Helper class providing Java version information
+ *
+ */
+public final class JavaVersion {
+
+	private static final String JAVA_VERSION;
+
+	private static final int MAJOR_VERSION;
+
+	static {
+		JAVA_VERSION = System.getProperty("java.version");
+		final String[] versionStrings = JAVA_VERSION.split("\\.");
```

Comment:
Can we write like,

```java
double version = Double.parseDouble(System.getProperty("java.specification.version"));
if(version > 1.8) {
//Do something
} else {
//Do something
}
```

Just curious, why? and why not? It eliminates the need of this Utility class.

Reply:
What is `Double.parseDouble("1.8.0")`?

Reply:
Seems like java.specification.version just returns 1.x or 11 (in cases of new versioning). I didn't know about this property. So IMO we could use this and get rid of the JavaVersion class.

Reply:
Ah, apologies @mohamedanees6, I didn't notice `java.specification.version` vs `java.version`.  That does look better to me, especially since `java.version` seems to have an unreliable format: https://stackoverflow.com/a/5103166/1153071

Your call @bender316 :)

Reply:
I will switch to the proposal above and remove the JavaVersion class.